File: arkham-horror-frontend/src/arkham/types/Phase.ts
import { JsonDecoder } from 'ts.data.json';

export type Phase = 'InvestigationPhase' | 'EnemyPhase' | 'UpkeepPhase' | 'MythosPhase';

export const phaseDecoder = JsonDecoder.oneOf<Phase>([
  JsonDecoder.isExactly('InvestigationPhase'),
  JsonDecoder.isExactly('EnemyPhase'),
  JsonDecoder.isExactly('UpkeepPhase'),
  JsonDecoder.isExactly('MythosPhase'),
], 'Phase');

File: backend/arkham-core/library/Arkham/Types/Phase.hs
module Arkham.Types.Phase
  ( Phase(..)
  ) where

import Arkham.Prelude

data Phase
  = MythosPhase
  | InvestigationPhase
  | EnemyPhase
  | UpkeepPhase
  | ResolutionPhase
  | CampaignPhase
  deriving stock (Eq, Ord, Show, Generic)
  deriving anyclass (ToJSON, FromJSON)
